File: 507 - Perfect Number.java
class Solution {
    public boolean checkPerfectNumber(int num) {
        int sum = 0;
        int[] x;
        for (int i = 1; i * i <= num; i++) {
            x = divmod(num, i);
            if (x[1] == 0) {
                sum += i;
                if (i != x[0]) {
                    sum += x[0];
                }
            }
        }
        sum -= num;
        return sum == num;
    }
    public int[] divmod(int n, int d) {
        return new int[] {(n / d), (n % d)};
    }
}

File: 507 - Perfect Number.py
import math
class Solution:
    def checkPerfectNumber(self, num: int) -> bool:
        divisor = set()
        divisor.add(num)
        s = 0
        for i in range(1, int(math.sqrt(num)) + 1):
            x, y = divmod(num, i)
            if y == 0:
                # divisor.add(i)
                s += i
                if i != x:
                    s += x
                # divisor.add(x)
        s -= num
        return num == s